report/map_helpers.go

```diff
@@ -138,3 +139,48 @@ func mapWrite(m ps.Map, encoder *codec.Encoder, encodeValue func(*codec.Encoder,
 	})
 	z.EncSendContainerState(containerMapEnd)
 }
+
+// Now follow helpers for StringLatestMap
+
+// These let us sort a StringLatestMap strings by key
+func (m StringLatestMap) Len() int           { return len(m) }
+func (m StringLatestMap) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+func (m StringLatestMap) Less(i, j int) bool { return m[i].key < m[j].key }
+
+func (m StringLatestMap) addMapEntries(ts time.Time, n map[string]string) StringLatestMap {
+	out := make(StringLatestMap, len(m), len(m)+len(n))
+	copy(out, m)
+	for k, v := range n {
+		out = append(out, stringLatestEntry{key: k, Value: v, Timestamp: ts})
+	}
+	out.fixup()
+	return out
+}
+
+func (m StringLatestMap) fixup() {
+	sort.Sort(m)
+	for i := 1; i < len(m); { // shuffle down any duplicates
+		if m[i-1].key == m[i].key {
+			if m[i-1].Timestamp.Before(m[i].Timestamp) {
+				m = append(m[:i-1], m[i:]...)
+			} else {
+				m = append(m[:i], m[i+1:]...)
+			}
+		} else {
+			i++
+		}
+	}
+}
```

Comment:
nitpick: I'd move this method up so it comes before the two methods that invoke it.

Reply:
moved.